models/stdp_triplet_connection.h

```diff
@@ -0,0 +1,355 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef STDP_TRIPLET_CONNECTION_H
+#define STDP_TRIPLET_CONNECTION_H
+
+/* BeginDocumentation
+  Name: stdp_triplet_synapse - Synapse type with spike-timing dependent
+  plasticity accounting for spike triplets as described in [1].
+
+  Description:
+    stdp_triplet_synapse is a connection with spike time dependent
+    plasticity accounting for spike triplet effects (as defined in [1]).
+
+  STDP examples:
+    pair-based   Aplus_triplet = Aminus_triplet = 0.0
+    triplet      Aplus_triplet = Aminus_triplet = 1.0
+
+  Parameters:
+    tau_plus           double: time constant of short presynaptic trace (tau_plus of [1])
+    tau_plus_triplet   double: time constant of long presynaptic trace (tau_x of [1])
+    Aplus              double: weight of pair potentiation rule (A_plus_2 of [1])
+    Aplus_triplet      double: weight of triplet potentiation rule (A_plus_3 of [1])
+    Aminus             double: weight of pair depression rule (A_minus_2 of [1])
+    Aminus_triplet     double: weight of triplet depression rule (A_minus_3 of [1])
+
+  States:
+    Kplus              double: pre-synaptic trace (r_1 of [1])
+    Kplus_triplet      double: triplet pre-synaptic trace (r_2 of [1])
+
+  Transmits: SpikeEvent
+
+  References:
+    [1] J.-P. Pfister & W. Gerstner (2006) Triplets of Spikes in a Model
+        of Spike Timing-Dependent Plasticity.  The Journal of Neuroscience
+        26(38):9673-9682; doi:10.1523/JNEUROSCI.1425-06.2006
+
+  Notes:
+    - Presynaptic traces r_1 and r_2 of [1] are stored in the connection as
+      Kplus and Kplus_triplet and decay with time-constants tau_plus and
+      tau_plus_triplet, respectively.
+    - Postsynaptic traces o_1 and o_2 of [1] are acquired from the post-synaptic
+      neuron states Kminus_ and triplet_Kminus_ which decay on time-constants
+      tau_minus and tau_minus_triplet, respectively. These two time-constants can
+      can be set as properties of the postsynaptic neuron.
+    - This version implements the 'all-to-all' spike interaction of [1]. The 'nearest-spike'
+      interaction of [1] can currently not be implemented without changing the postsynaptic
+      archiving-node (clip the traces to a maximum of 1).
+
+  FirstVersion: Nov 2007
+  Author: Moritz Helias, Abigail Morrison, Eilif Muller, Alexander Seeholzer, Teo Stocco
+  SeeAlso: synapsedict, stdp_synapse, static_synapse
```

Comment:
Ah, sorry, my bad. I wasn't aware of the double registering in ``modelsmodule``.
Will add the reference.